src/System.Private.DataContractSerialization/src/System/Xml/XmlUTF8TextWriter.cs

```diff
@@ -101,7 +101,7 @@ public XmlUTF8NodeWriter(bool[] isEscapedAttributeChar, bool[] isEscapedElementC
         new public void SetOutput(Stream stream, bool ownsStream, Encoding encoding)
         {
             Encoding utf8Encoding = null;
-            if (encoding != null && encoding == Encoding.UTF8)
+            if (encoding != null && encoding.CodePage == Encoding.UTF8.CodePage)
```

Comment:
The Equals method also checks encoder/decoder fallback of the encoding instance besides checking the CodePage, which is different from this check. This is the behavior we have on Desktop. I'd prefer to keep the change as is for consistency.